### PR TITLE
Improved search.py

### PR DESCRIPTION
**Pull request by louwersj**
made several improvements to the search.py codebase

### CODE REVIEW

**Comment by hulsed** [comment]
See comment. Overall, search.py is not in a working state anymore/yet with this alpha version so we may have some changes in the pipeline in the next few weeks as we bring the examples up to spec.

**Comment by louwersj** [comment]
@hulsed  could you still merge this imporved / cleaned one in the project main 

We could properly forked it from that point onwards.

File: fmdtools/define/block.py [hunk 1]
```diff
@@ -1048,37 +1046,14 @@ class FxnBlock(Block):
     Superclass class for functions which is a special type of Block\
     with c and a attributes for CompArch and ASGs, as well as a defined method for propagation
     """
-    def __init__(self, name='', flows={}, c=dict(), a=dict(), local=dict(), args_f=dict(), **kwargs):
+    def __init__(self, name='', flows={}, args_f=dict(), **kwargs):
```
Comment (hulsed) [inline fmdtools/define/block.py:1049]: This is removing too many attributes and will break certain models/workflows that rely on the ASG and CompArch types:
- c and a are instantiated just a bit below, so they need to be kept here
- local needs to be kept (and I think is supposed to call add_local_to_flowdict, which I think is supposed to happen in comms also--so, would make a good issue/bug report)
- The previous docs for parameters is mostly accurate--p,s,r,m,t, and sp are all parameters that get passed as kwargs to Block. Maybe we need a better way of documenting?

**Review by hulsed — CHANGES_REQUESTED** [review]
Need to adjust some of the changes to FxnBlock

File: fmdtools/define/block.py (hunk 1 above)
Comment (hulsed) [reply fmdtools/define/block.py:1049]: It looks like your latest commit only has the changes to search.py--make sure this makes it in and I'll merge the PR

**Comment by hulsed** [comment]
> @hulsed changes are made to correct the requested. Can you have a quick check on the two changes done to search.py and block.py ?

I don't see that the corrections to block.py made it in but will once they are there

**Comment by louwersj** [comment]
adding corrections to block.py @hulsed

**Review by hulsed — APPROVED** [review]

**Pull request merged**